app/controllers/premiumlease/CalculatedFigureYourselfController.scala

```diff
@@ -64,7 +65,16 @@ class CalculatedFigureYourselfController @Inject()(
           for {
             updatedAnswers <- Future.fromTry(request.userAnswers.set(CalculatedFigureYourselfPage, value))
             _              <- sessionRepository.set(updatedAnswers)
+            _              = if (value.calculatedFigureYourself) clearData
```

Comment:
Instead of doing this, we can follow the cleanUp pattern from the pages. [Here](https://github.com/hmrc/income-tax-property-frontend/blob/main/app/pages/propertyrentals/IsNonUKLandlordPage.scala#L31) is an example that we implemented recently, and that way we leave the responsibility separated.
Also, this can be tested separately, see the example [here](https://github.com/hmrc/income-tax-property-frontend/blob/main/test/pages/propertyrentals/IsNonUKLandlordPageSpec.scala#L23)